src/containers/Home/components/ProductForm/index.jsx

```diff
@@ -0,0 +1,178 @@
+import React, { useState } from 'react';
+import { toast } from 'react-hot-toast';
+import { useProductContext } from '../../../../contexts/ProductContext';
+import Form from '../MainForm';
+
+const ProductForm = () => {
+  const [showAddForm, setShowAddForm] = useState(false);
+  const [showEditForm, setShowEditForm] = useState(false);
+  const [fetched, setFetched] = useState(false);
+  const { state, dispatch } = useProductContext();
+  const { products } = state;
+  const [formData, setFormData] = useState({
+    title: '',
+    price: '',
+    description: '',
+    image: null,
+  });
+
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+    e.target.classList.contains('is-invalid') &&
+      e.target.classList.remove('is-invalid');
+    e.target.classList.contains('is-valid') &&
+      e.target.classList.remove('is-valid');
+    if (name === 'price' && value <= 0) {
+      e.target.classList.add('is-invalid');
+      toast.error('Price must be greater than 0.');
+      return;
+    } else if (value === '') {
+      e.target.classList.add('is-invalid');
+      toast.error('Please fill in all fields.');
+      return;
+    }
+    e.target.classList.add('is-valid');
+    setFormData({ ...formData, [name]: value });
+  };
+
+  const handleImageChange = (e) => {
+    const file = e.target.files[0];
+    e.target.classList.contains('is-invalid') &&
+      e.target.classList.remove('is-invalid');
+    e.target.classList.contains('is-valid') &&
+      e.target.classList.remove('is-valid');
```

Comment:
We are repeating these operations in both handleChange & handleImageChange functions. We can move them to separate function.